src/io/iohk/scalanet/peergroup/TCPPeerGroup.scala

```diff
@@ -48,6 +48,16 @@ class TCPPeerGroup[F[_]](val config: Config)(implicit liftF: Lift[F])
     .bind(config.bindAddress)
     .syncUninterruptibly()
 
+  private val subscribers = new Subscribers[ByteBuffer]()
+
+  override val messageStream: MessageStream[ByteBuffer] = subscribers.messageStream
+
+  override val processAddress: InetSocketAddress = config.processAddress
+
+  messageStream.foreach { byteBuffer =>
+    Codec.decodeFrame(decoderTable.entries, 0, byteBuffer)
```

Comment:
Is this the only way to do this? I find trivial to miss this detail, also, is this behavior even tested?

Reply:
Yes, it's tested. Typed messaging breaks without it.

We're looking at whether this is the right code in another story.